src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs

```diff
@@ -1148,5 +1198,121 @@ public async Task Pound_r_nuget_should_display_only_requested_packages(Language
                   .And
                   .NotContainAny(notExpectedDisplayed);
         }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        [InlineData(Language.FSharp)]
+        public async Task Pound_r_nuget_should_display_only_requested_packages_for_third_submission(Language defaultLanguageKernel)
+        {
+            var kernel = CreateCompositeKernel(defaultLanguageKernel);
+
+            var codeFirstSubmission = @"
+#r ""nuget:Microsoft.ML.OnnxTransformer,1.4.0""
+";
+            var command1 = new SubmitCode(codeFirstSubmission);
+            var _result1 = await kernel.SendAsync(command1);
+
+            var codeSecondSubmission = @"
+#r ""nuget:Google.Protobuf, 3.5.1""
+";
+            var command2 = new SubmitCode(codeSecondSubmission);
+            var result2 = await kernel.SendAsync(command2);
+
+            var codeThirdSubmission = @"
+#r ""nuget:Google.Protobuf, 3.5.1""
+#r ""nuget:Microsoft.ML.OnnxTransformer, 1.4.0""
+";
+            var command3 = new SubmitCode(codeThirdSubmission);
+            var result3 = await kernel.SendAsync(command3);
+
+            using var events = result3.KernelEvents.ToSubscribedList();
```

Comment:
I would subscribe to the event before submitting the command that is to be tested

Reply:
updated